rfcs/react-components/convergence/custom-styling.md

```diff
@@ -0,0 +1,214 @@
+
+For example, imagine this Button.tsx is defined in a partner component library to export a custom styled button.
+The component exported is named `Button` so that it can replace Fluent UI's `Button`.
+
+The partner would define the useCustomButtonStyles hooks to style the component however they like.
+This could include exposing a similar centralized custom styling mechanism as in Option A or applying styles
+through a different mechanism than Griffel (such as SASS, emotion, or tailwind).
+
+```tsx
+import * as React from 'react';
+import { renderButton_unstable, useButton_unstable } from '@fluentui/react-components';
+import type { ButtonProps, ForwardRefComponent } from '@fluentui/react-components';
+import { useCustomButtonStyles } from '../styling/button';
+
+export const Button: ForwardRefComponent<ButtonProps> = React.forwardRef((props, ref) => {
+  const state = useButton_unstable(props, ref);
+  useCustomButtonStyles(state);
+  return renderButton_unstable(state);
+}) as ForwardRefComponent<ButtonProps>;
+
+Button.displayName = 'Button';
+```
+
+### Improvement - making recomposition easier with compose()
+
+Recomposing every component could be made easier with a compose method that hid some of the details of the ForwardRef
+and sequencing the hook.
+
+```tsx
+import * as React from 'react';
+import { compose } from '@fluentui/react-components';
+import type { ButtonProps } from '@fluentui/react-components';
+import { useCustomButtonStyles } from '../styling/button';
+
+// This is a rough idea of what a compose method might look like.
+export const Button = compose<ButtonProps>('Button', useButton_unstable, useCustomButtonStyles, renderButton_unstable);
+```
+
+> Ben Howell is working on a proof-of-concept for this compose method.
+> It is likely something worth doing regardless of the outcome of this RFC.
+> The compose method could be called from our own components to reduce boilerplate code.
+
+### Improvement - generating a recomposed package
+
+We could create an nx generator that produces all the boilerplate component recomposition code.
+This would allow partners to create an unstyled or custom library easily.
+Partners could re-run the tool to keep up with new components shipping in react-components.
+
+```
+yarn create-recomposed-library my-components
+```
+
+### Open issues
+
+### Pros
+
+- doesn't add yet another way to customize styles to the architecture
+- uses the hooks composition model for its intended purpose, to handle special cases of customization
+- partners get complete control over customization (vs. limitations that may come from a style-only customizer)
+- allows replacement of Griffel with a different styling system
+- supports tree-shaking out the hooks that are not used
+- follows out recommended dependency abstraction approach for partners with large and complex code bases to make future migration easier.
+
+### Cons
+
+- does not provide partners with an out-of-the-box solution
```

Comment:
These two are big cons for our consumers. I see so many teams leveraging the analogous system from v8 that we're forcing engineering teams to each own a lot of boilerplate that Fluent has historically owned.